### DISCUSSION TITLE
Buttonpage button layout

### DISCUSSION THREAD

**Discussion by denes44** [Blackymas/NSPanel_HA_Blueprint#738] [Ideas]
I have a guestion/suggestion about the buttonpage layout.

This is the current layout of a button:
![nspanel_button](https://user-images.githubusercontent.com/60078357/234696452-4873f6b4-a3b6-4c69-bb6e-5205236239a3.jpg)
Is anything planned to the now empty space, or is there another reason that the space between the icon+value and label is left?

I'm asking because 10 characters is not very much, and it would be good if the buttons supports two rows of text.

Since the wordwrap functionality of the nextion display is very bad, what can be done is to resize the textbox to fit two rows of text, and align the text vertically to the bottom. That way it's backwards compatible, because if there is one row of text, it will appear the same, one row of text at the bottom of the button.
Since we cannot use the built in wordwrap functionality, the only way to utilize the two rows is to use a custom label in the blueprint.

If I enlarge the textbox, and enter `Auto\rLights` as label in the blueprint, I got this:
![nspanel_button1](https://user-images.githubusercontent.com/60078357/234708756-58743f3d-aa16-4406-bd2c-f12e7109d8f4.jpg)
Unfortunately the horizontal center align is working in a somewhat special way in this display:
![nspanel_button2](https://user-images.githubusercontent.com/60078357/234708759-901299f9-7dbc-4a2a-b8bf-b87029b6bdef.jpg)

But I think it's still better than nothing if you cannot fit inside the 10 character limit. And it also does not do any harm if somebody doesn't want to use two lines.

**Comment by denes44** [comment]
I created a PR for this: #740

**Comment by rleongcs** [comment]
Exactly what I was going to ask if we can have two lines of text as I am using lots of abbreviations lately 😉

**Comment by edwardtfn** [comment]
I'm closing this discussion as we have released v4.0 (#1057).
Please add a feature request for the missing things.

And thanks a lot for your support.
